### DISCUSSION TITLE
devcontainer: How To Access Forwarded Port With Another Device On The Same Network

### DISCUSSION THREAD

**Discussion by aaljukic** [community/community#13473] [Codespaces] [answered]
- VSCode Version: 1.65.2
- Local OS Version: macOS 12.2.1 (21D62)
- Local chip architecture: Apple Silicon
- Docker Desktop 4.6.1 (76265)
- Reproduces in: Remote - Containers

### Is this a bug?

Not sure, but it looks like one

### Problem

Can't connect with a phone to the server (forwarded port) on the same local network. 

### Question

I've recently started doing everything inside dev containers (must say, it's amazing) but one thing I can't figure out is how to access the server (port) from another machine on the same network. Maybe I'm missing something. It should be accessible because it is forwarded to the host machines port but it doesn't work. 

I'm using a mac so the firewall is disabled by default but also if I use a built-in mac apache server I can access port 80 without any problems. I can also ping devices so I will exclude network or firewall issues. Also, I've tried without docker on another machine and it does work. 

I tried to change Docker Subnet inside Docker preferences > resources > network to correspond to the hosts subnet but yet no progress. I also tested with multiple docker images like ubuntu, node, and node:17-alpine and with multiple different projects ViteJs, Create React App and with just serving it through npm serve. When I add --network=host, I can see that it is using correct subnet instead of 172.17.0.x but still I can't reach site. 

Sorry if this is not the right place to ask this question but I couldn't find anything on this subject 

### Steps to reproduce

devcontainer.json
```
{
  "name": "Test",
  "runArgs": [
      "--name=test",
      "--network=host"
  ],
  "build": {
      "dockerfile": "Dockerfile",
      "args": {
        "network": "host",
      },
  },
}
```

Dockerfile
`FROM node:17-alpine`

```
npm create vite@latest
cd vite-project
```

add --host 0.0.0.0 to dev script

`npm run dev`

Try to access ip address with another device on the same network

**Comment by aaljukic** [comment, marked as answer]
Ok, i managed to solve it. I removed --network=host from runArgs and "network": "host" from build.args and added 
"-p=192.168.x.x:3000:3000" to runArgs

**Reply by rafael-perini** [reply]
It seems to me that you can even remove the `192.168.x.x:`.
At least for me it worked just with `"runArgs": ["-p=3000:3000"]`.

**Reply by LuisHenriqueJung** [reply]
Thanks a lot @aaljukic

**Comment by mmghv** [comment]
For me what worked is to change this setting in VSCode remote extension:

> Remote: Local Port Host
> allInterfaces

Source :
https://stackoverflow.com/questions/67983378/specify-listening-address-0-0-0-0-for-forwarded-port-in-vscode-remote-containers

**Comment by ashrafmo-1** [comment]
rate #1
